### PR TITLE
Fix: allows one to login with invalid skills/reps

### PR DESCRIPTION
**Pull request by fleisch11**
it also makes invalid skills not show up at all, **but they never get deleted**

### CODE REVIEW

File: BondageClub/Scripts/Reputation.js [hunk 1]
```diff
@@ -41,9 +41,10 @@ function ReputationChange(RepType, RepValue, Push) {
 	}
 
 	else {
-		throw new Error("trying to change invalid reputation")
+		if (CurrentScreen != "Login") {
+			throw new Error("trying to change invalid reputation")
```
Comment (elliesec) [inline BondageClub/Scripts/Reputation.js:45]:
```suggestion
			console.warn(`Invalid reputation type "${RepType}"`);
```
I didn't have a proper look at the previous PR before it was merged, so didn't notice this then, but I feel like throwing errors here is perhaps a little harsh. I don't feel that using an invalid reputation type should be game-breaking, but we should definitely be warning about it. 

I'd suggest maybe switching them to use a `console.warn`. Including the invalid type in the message might also be useful information for future devs, particularly if they want to add new reputation/skill types.

File: BondageClub/Scripts/Skill.js [hunk 2]
```diff
@@ -45,7 +45,9 @@ function SkillChange(SkillType, SkillLevel, SkillProgress, Push) {
 	}
 
 	else {
-		throw new Error("trying to change invalid skill");
+		if (CurrentScreen != "Login") {
+			throw new Error("trying to change invalid skill");
```
Comment (elliesec) [inline BondageClub/Scripts/Skill.js:49]:
```suggestion
			console.warn(`Invalid skill type "${SkillType}"`);
```
Same comment here as in `Reputation.js`.

**Comment by fleisch11** [comment]
@elliesec allright, i will remove the login stuff too, once the repostery has downloaded

**Comment by fleisch11** [comment]
@elliesec alright, everything is done now

**Review by elliesec — APPROVED** [review]

**Comment by Ben987** [comment]
Absolutely!  Thanks.

**Pull request merged**